src/imagery/i.eodag/i.eodag.py

```diff
@@ -799,6 +829,47 @@ def products_compare(first, second):
     return search_result
 
 
+def skip_existing(output, search_result):
+    """Remove products that is already downloaded and saved in 'output' directory.
+
+    :param output: Output directory whose files will be compared with the scenes.
+    :type output: class'eodag.api.search_result.SearchResult'
+
+    :param search_results: EO products to be checked for existence in 'output' directory.
+    :type search_result: class'eodag.api.search_result.SearchResult'
+
+    :return: Sorted EO products
+    :rtype: class:'eodag.api.search_result.SearchResult'
+    """
+    filtered_result = []
+    # Check for previously downloaded scenes
+    existing_files = [os.path.join(output, f) for f in os.listdir(output)]
+    if len(existing_files) <= 1:
```

Comment:
```suggestion
    if existing_files:
```
If a single file exists in _output_ it should be checked, no? And if you check for an empty list, don`t use _len()_